src/rapidquilt/apply/common.rs

```diff
@@ -158,8 +163,11 @@ pub fn save_modified_file<'arena, H: BuildHasher>(
         let mut output = File::create(file_path)?;
 
         // If any patch set non-default permission, set them now
+        // Otherwise preserve existing permissions
         if let Some(ref permissions) = file.permissions {
             output.set_permissions(permissions.clone())?;
+        } else if let Some(ref permissions) = old_permissions {
+            output.set_permissions(permissions.clone())?;
```

Comment:
I don't like repeating `output.set_permissions`. Why not use `or`? That is:
```
if let Some(ref permissions) = file.permissions.or(old_permissions) {
```

Reply:
Erm, I have just checked, and my proposed code does not compile as such. You would have to use `as_ref()`. But since the permissions must be later cloned for `set_permissions()`, I suggest to write something like:
```
if let Some(permissions) = file.permissions.clone().or(old_permissions) {
    output.set_permissions(permissions)?;
```